src/flush-microtasks.js

```diff
@@ -1,3 +1,5 @@
+import scheduleCallback from './scheduler-compat'
```

Comment:
I don't think having the compat file is necessary.

```suggestion
import React from 'react'
import semver from 'semver'

const globalObj = typeof window === 'undefined' ? global : window

let Scheduler = globalObj.Scheduler

const isModernScheduleCallbackSupported = semver.satisfies(React.version, '>=16.8.6')
```

Also, let's move this code to below the comments, just above the `getIsUsingFakeTimers`

Reply:
You're definitely right about this one, I thought we don't want to use `semver`.. This was my go-to solution but since `act-compat` isn't using it and I didn't see it in the project, I didn't want to add it.
This simplifies everything.